src/lib/utils.spec.js

```diff
@@ -178,3 +189,53 @@ describe('getUserNameFromUrl', () => {
     expect(getUserNameFromUrl()).toBeNull()
   })
 })
+
+describe('generateReturnUrlToNotesIndex', () => {
+  function createFetchUrl() {
+    return jest.fn().mockImplementation(({ _id }) => ({
+      data: {
+        type: 'io.cozy.notes.url',
+        id: _id,
+        note_id: _id,
+        subdomain: 'flat',
+        protocol: 'https',
+        instance: 'alice.cozy.example',
+        public_name: 'Bob'
+      }
+    }))
+  }
+
+  function createClient() {
+    return {
+      getStackClient: () => ({
+        collection: () => ({ fetchURL: createFetchUrl() })
+      })
+    }
+  }
+
+  function createNote(id = '12345') {
+    return { id, type: 'io.cozy.files' }
+  }
+
+  afterEach(() => {
+    restoreLocation()
+  })
+
+  it('returns an URL string', async () => {
+    const client = createClient()
+    const note = createNote()
+    const urlString = await generateReturnUrlToNotesIndex(client, note)
+    const url = new URL(urlString)
+    expect(url.toString()).toEqual(urlString)
```

Comment:
When can this test fail? SInce we compare the result of the function with itself, I don't think it can? It probably covers the case where `new URL()` fails wut it's not very explicit.

Reply:
Yep, it'll fail if `url` is not an URL. That was the purpose